src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -980,7 +980,7 @@ impl EventProcessor {
     {
         let wt = Self::window_target(&self.target);
         let window_id = mkwid(event.event as xproto::Window);
-        let device_id = mkdid(event.deviceid as xinput::DeviceId);
+        let device_id = mkdid(event.sourceid as xinput::DeviceId);
```

Comment:
For all of these usages it should be indicated in comments that `sourceid` is only supported since Xinput v2.1.